test/library/packages/LinearAlgebra/correctness/no-dependencies/testExpm.chpl

```diff
@@ -0,0 +1,459 @@
+use LinearAlgebra;
+use TestUtils;
+
+{
+  const vec = 0..1;
+  var D = {vec,vec};
+  var X = eye(D);
+  X(0,1) = 1;
+  X = reshape(2*X, D);
+
+  var E = expm(X);
+
+  X(0,0) = 7.389056;
+  X(0,1) = 14.778112;
+  X(1,0) = 0.0;
+  X(1,1) = 7.389056;
+
+  assertAlmostEqual(E, X, 'Matrix Exponential - X = 2*2');
+}
+
+{
+  const vec = 0..2;
+  var D = {vec,vec};
+  var X = eye(D);
+  X(0,1) = 1;
+  X(0,2) = 1;
+  X(2,0) = 1;
+  X = reshape(2*X, D);
+
+  var E = expm(X);
+
+  X(0,0) = 27.799075;
+  X(0,1) = 26.799075;
+  X(0,2) = 26.799075;
+  X(1,0) = 0.0;
+  X(1,1) = 7.389056;
+  X(1,2) = 0.0;
+  X(2,0) = 26.799075;
+  X(2,1) = 20.410019;
+  X(2,2) = 27.799075;
+
+  assertAlmostEqual(E, X, 'Matrix Exponential - X = 3*3');
+}
+
+{
+  //Pade3 case
+  const vec = 0..2;
+  var D = {vec,vec};
+  var X = eye(D);
+  var w = 0.01;
+  var Y = X * w;
+
+  var E = expm(Y);
+
+  X(0,0) = 1.010050;
+  X(0,1) = 0.0;
+  X(0,2) = 0.0;
+  X(1,0) = 0.0;
+  X(1,1) = 1.010050;
+  X(1,2) = 0.0;
+  X(2,0) = 0.0;
+  X(2,1) = 0.0;
+  X(2,2) = 1.010050;
+
+  assertAlmostEqual(E, X, 'Matrix Exponential - X = 3*3 (Uses Pade3)');
+}
+
+{
+  //Pade5 case
+  const vec = 0..2;
+  var D = {vec,vec};
+  var X = eye(D);
+  var w = 0.1;
+  var Y = X * w;
+
+  var E = expm(Y);
+
+  X(0,0) = 1.105171;
+  X(0,1) = 0.0;
+  X(0,2) = 0.0;
+  X(1,0) = 0.0;
+  X(1,1) = 1.105171;
+  X(1,2) = 0.0;
+  X(2,0) = 0.0;
+  X(2,1) = 0.0;
+  X(2,2) = 1.105171;
+
+  assertAlmostEqual(E, X, 'Matrix Exponential - X = 3*3 (Uses Pade5)');
+}
+
+{
+  //Pade7 case
+  const vec = 0..2;
+  var D = {vec,vec};
+  var X = eye(D);
+  var w = 0.5;
+  var Y = X * w;
+
+  var E = expm(Y);
+
+  X(0,0) = 1.648721;
+  X(0,1) = 0.0;
+  X(0,2) = 0.0;
+  X(1,0) = 0.0;
+  X(1,1) = 1.648721;
+  X(1,2) = 0.0;
+  X(2,0) = 0.0;
+  X(2,1) = 0.0;
+  X(2,2) = 1.648721;
+
+  assertAlmostEqual(E, X, 'Matrix Exponential - X = 3*3 (Uses Pade7)');
+}
+
+{
+  //Pade9 case
+  const vec = 0..2;
+  var D = {vec,vec};
+  var X = eye(D);
+  var w = 1;
+  var Y = X * w;
+
+  var E = expm(Y);
+
+  X(0,0) = 2.718282;
+  X(0,1) = 0.0;
+  X(0,2) = 0.0;
+  X(1,0) = 0.0;
+  X(1,1) = 2.718282;
+  X(1,2) = 0.0;
+  X(2,0) = 0.0;
+  X(2,1) = 0.0;
+  X(2,2) = 2.718282;
+
+  assertAlmostEqual(E, X, 'Matrix Exponential - X = 3*3 (Uses Pade9)');
+}
+
+{
+  //Pade13 case
+  const vec = 0..2;
+  var D = {vec,vec};
+  var X = eye(D);
+  var w = 10;
+  var Y = X * w;
+
+  var E = expm(Y);
+
+  X(0,0) = 22026.465795;
+  X(0,1) = 0.0;
+  X(0,2) = 0.0;
+  X(1,0) = 0.0;
+  X(1,1) = 22026.465795;
+  X(1,2) = 0.0;
+  X(2,0) = 0.0;
+  X(2,1) = 0.0;
+  X(2,2) = 22026.465795;
+
+  assertAlmostEqual(E, X, 'Matrix Exponential - X = 3*3 (Uses Pade13)');
+}
+
+{
```

Comment:
I would write this sequence of tests (for different values of b) as a loop, and the checked result as an expression. I think that will make this clearer, for anyone reading this down the line.